pkg/controller/instance/instance_controller.go

```diff
@@ -265,49 +267,39 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 					}
 				}
 
-				if err = createPlan(mgr, planName, new); err != nil {
+				if err = createPlanOld(mgr, planName, new); err != nil {
 					log.Printf("InstanceController: Error creating PlanExecution \"%v\" for instance \"%v\": %v", planName, new.Name, err)
 				}
 			}
 
 			// See if there's a current plan being run, if so "cancel" the plan run.
 			return e.ObjectOld != e.ObjectNew
 		},
-		// New Instances should have Deploy called
+		// New Instances should confirm there is a deployment plan without side-effects)
 		CreateFunc: func(e event.CreateEvent) bool {
 			log.Printf("InstanceController: Received create event for instance \"%v\"", e.Meta.GetName())
+			ctx := context.TODO()
+
+			// In order to create we must have an OV and it must have a "deploy" plan.
 			instance := e.Object.(*kudov1alpha1.Instance)
 
-			// Get the instance OperatorVersion object
-			ov := &kudov1alpha1.OperatorVersion{}
-			err := mgr.GetClient().Get(context.TODO(),
-				types.NamespacedName{
-					Name:      instance.Spec.OperatorVersion.Name,
-					Namespace: instance.GetOperatorVersionNamespace(),
-				},
-				ov)
-			if err != nil {
+			ov, err := getOperatorVersion(ctx, mgr.GetClient(), nil, instance)
```

Comment:
I feel like we do exactly the same thing in reconcile, why would we not rely just on reconcile dealing with this? We pull OV there in the beginning so we could have this logic just in one place

Reply:
the use of OV is different in the 2 different places...
here:  the OV is checked to see if a deploy plan exists where a false is return if it isn't... thus indicating that the create event should not be processed.  
in reconcile: OV is used to actually create or exec the plan (and associate it with the instance)